packages/react-components/react-menu/library/src/components/MenuItem/useMenuItemStyles.styles.ts

```diff
@@ -175,6 +187,63 @@ const useStyles = makeStyles({
   },
 });
 
+const useMultilineStyles = makeStyles({
+  root: {
+    display: 'grid',
+    gridTemplateColumns: `var(${cssVars.menuItemMultilineGridColumns})`,
+    gridTemplateAreas: `var(${cssVars.menuItemMultilineGridAreas})`,
+  },
+
+  checkmark: {
+    gridArea: 'checkmark',
+  },
+
+  icon: {
+    gridArea: 'icon',
+  },
+
+  content: {
+    gridArea: 'content',
+  },
+
+  secondaryContent: {
+    gridArea: 'secondaryContent',
+    alignSelf: 'center',
+  },
+
+  subText: {
+    gridArea: 'subText',
+  },
+
+  submenuIndicator: {
+    alignSelf: 'center',
+    gridArea: 'submenuIndicator',
+  },
+});
+
+const buildMultilineGridArea = (slots: MenuItemSlots, hasSubmenu: boolean) => {
+  const checkmark = slots.checkmark ? 'checkmark' : '';
+  const icon = slots.icon ? 'icon' : '';
+  const secondaryContent = slots.secondaryContent ? 'secondaryContent' : '';
+  const submenuIndicator = slots.submenuIndicator || hasSubmenu ? 'submenuIndicator' : '';
```

Comment:
use this to avoid the typical 'gaps still apply even if nothing is in the grid area' problem